src/Nethermind/Nethermind.Merge.Plugin/MergePlugin.cs

```diff
@@ -287,7 +316,13 @@ public Task InitSynchronization()
             return Task.CompletedTask;
         }
 
-        public ValueTask DisposeAsync() => ValueTask.CompletedTask;
+        public ValueTask DisposeAsync()
+        {
+            return new ValueTask(Task.Run(() =>
+            {
+                _invalidChainTracker?.Dispose();
+            }));
```

Comment:
just do `_invalidChainTracker?.Dispose(); return default;` you won't be allocating new Task on heap.

Reply:
Ok